File: lib_internet/src/main/java/world/share/lib_internet/http/exception/FactoryException.java
package world.share.lib_internet.http.exception;


/**
 * 错误工厂
 *
 * @author mac
 */
public class FactoryException {

    /**
     * 封装错误
     *
     * @param e 错误
     * @return ApiException    自定义错误
     */
    public static ApiException analysisExcetpion(Throwable e) {
        ApiException apiException = new ApiException(e);
        return apiException;
    }
}

File: lib_base/src/main/java/world/share/lib_base/internet/http/exception/FactoryException.kt
package world.share.lib_base.internet.http.exception

/**
 * 错误工厂
 *
 * @author mac
 */
object FactoryException {
    /**
     * 封装错误
     *
     * @param e 错误
     * @return ApiException    自定义错误
     */
    fun analysisExcetpion(e: Throwable?): ApiException {
        return ApiException(e)
    }
}